### ISSUE TITLE
Rule Change: object-curly-spacing may not apply to import/export declarations

### ISSUE THREAD

**Issue by iabervon** [eslint/eslint#15711]
### What rule do you want to change?

object-curly-spacing

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
import { name } from './modules/square.js';

let x = {greeting: name};
```

### What does the rule currently do for this code?

Generates a warning of one sort or another, either for { name } or for {greeting: name}, depending on which way the string option is set.

### What will the rule do after it's changed?

Not generate a warning if the object option is set to expect this combination.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

The example is parts of two MDN tutorial examples, one for modules and one for object initializers. It seems like Mozilla's style guide considers braces around lists of names more like blocks and less like object destructuring (note that it also would have block-spacing: always, based on other examples).

I'm not sure what the right way to set up option(s) for this is, given that the variation between style guides seems to be in which of two rules applies to some syntax. I think just having a "moduleDeclarations": boolean option here would be sufficient, without getting into what style guide is matching instead of usual objects.

**Comment by TriMoon** [comment]
@iabervon 
>I think just having a "moduleDeclarations": boolean option [here](https://eslint.org/docs/rules/object-curly-spacing) would be sufficient, without getting into what style guide is matching instead of usual objects.

I agreed on the need, but not on the option choice.
IMHO it would be better to have a new separate `module-declarations` option or the like because there are more issues w.r.t. `import/export` keywords in particular.
(see https://github.com/eslint/eslint/issues/15710 for example...)

**Comment by mdjermanovic** [comment]
Hi @iabervon, thanks for the issue!

Per our [policies](https://github.com/eslint/eslint#stylistic-rule-updates), stylistic rules are frozen so we won't be adding any new options to the `object-curly-spacing` rule. If you'd like to enforce different formatting for different constructs, you can always copy the existing rule into a custom rule and make it behave the way you want.
